iceoryx_utils/test/moduletests/test_cxx_vector.cpp

```diff
@@ -1034,3 +1189,9 @@ TEST_F(vector_test, PartiallyEqualVectorsWithDifferentCapacityAreNotEqual)
     EXPECT_FALSE(a == b);
     EXPECT_TRUE(a != b);
 }
+
+TEST_F(vector_test, MemoryEfficientImplementation)
```

Comment:
Very nice, I like it. I think we should do this more often - creating tests with some efficiency requirements.